src/index.js

```diff
@@ -40,33 +57,65 @@ module.exports = class Client {
     this[PRIVATE.OFFSETS] = new Map()
     this[PRIVATE.LOGGER] = createLogger({ level: logLevel, logCreator })
     this[PRIVATE.CLUSTER_RETRY] = retry
-    this[PRIVATE.CREATE_CLUSTER] = ({
-      metadataMaxAge,
+    this[PRIVATE.CREATE_BROKERPOOL] = ({
+      metadataMaxAge = DEFAULT_METADATA_MAX_AGE,
       allowAutoTopicCreation = true,
       maxInFlightRequests = null,
       instrumentationEmitter = null,
-      isolationLevel,
-    }) =>
-      new Cluster({
+    }) => {
+      const connectionBuilder = createConnectionBuilder({
         logger: this[PRIVATE.LOGGER],
         retry: this[PRIVATE.CLUSTER_RETRY],
-        offsets: this[PRIVATE.OFFSETS],
+        instrumentationEmitter,
         socketFactory,
         brokers,
         ssl,
         sasl,
         clientId,
         connectionTimeout,
-        authenticationTimeout,
-        reauthenticationThreshold,
         requestTimeout,
         enforceRequestTimeout,
-        metadataMaxAge,
-        instrumentationEmitter,
-        allowAutoTopicCreation,
         maxInFlightRequests,
+      })
+      return new BrokerPool({
+        logger: this[PRIVATE.LOGGER],
+        retry: this[PRIVATE.CLUSTER_RETRY],
+        connectionBuilder,
+        allowAutoTopicCreation,
+        authenticationTimeout,
+        reauthenticationThreshold,
+        metadataMaxAge,
+      })
+    }
+    this[PRIVATE.CREATE_CLUSTER] = ({
+      isolationLevel,
+      brokerPool,
+      instrumentationEmitter,
+      ...brokerPoolOptions
+    }) => {
+      if (
```

Comment:
This part could be a lot clearer. I guess this is trying to determine if we're trying to create a cluster with options that are not compatible with the existing broker pool, but I can't really tell what's going on.

Reply:
I'll try to pull this one out, indeed it's quite a mess to follow.

Reply:
Moved it into a separate function, and tried to shorten the code a bit more to be easier to follow.